caas/clientconfig/k8s.go

```diff
@@ -0,0 +1,131 @@
+package caas
+
+import (
+	"os"
+	"path/filepath"
+
+	"k8s.io/client-go/tools/clientcmd"
+	clientcmdapi "k8s.io/client-go/tools/clientcmd/api"
+
+	"github.com/juju/errors"
+	"github.com/juju/loggo"
+
+	"github.com/juju/juju/cloud"
+)
+
+var logger = loggo.GetLogger("juju.caas.clientconfig")
+
+type K8SClientConfigReader struct {
+}
+
+func (r *K8SClientConfigReader) GetClientConfig() (*ClientConfig, error) {
```

Comment:
All exported symbols need a docstring (which starts with the name).

For example: `// GetClientConfig parses Kubernetes client configuration from the default location or $KUBECONFIG.`